certbot/account.py

```diff
@@ -156,22 +161,41 @@ def _key_path(cls, account_dir_path):
     def _metadata_path(cls, account_dir_path):
         return os.path.join(account_dir_path, "meta.json")
 
-    def find_all(self):
+    def _find_all_for_server_path(self, server_path):
+        accounts_dir = self.config.accounts_dir_for_server_path(server_path)
         try:
-            candidates = os.listdir(self.config.accounts_dir)
+            candidates = os.listdir(accounts_dir)
         except OSError:
             return []
 
         accounts = []
         for account_id in candidates:
             try:
-                accounts.append(self.load(account_id))
+                accounts.append(self._load_for_server_path(account_id, server_path))
             except errors.AccountStorageError:
                 logger.debug("Account loading problem", exc_info=True)
+
+
+        if not accounts and server_path in constants.LE_REUSE_SERVERS:
+            # find all for the next link down
+            prev_server_path = constants.LE_REUSE_SERVERS[server_path]
+            prev_accounts = self._find_all_for_server_path(prev_server_path)
+            # if we found something, link to that
+            if prev_accounts:
+                if os.path.islink(accounts_dir):
+                    os.unlink(accounts_dir)
+                else:
+                    os.rmdir(accounts_dir)
```

Comment:
This will fail if the directory isn't empty which could happen if the directory got corrupted somehow (which we handle gracefully above when loading accounts).

I'm open to other ideas, but if this fails, how about we return an empty list? In this corner case, the account won't be reused, but at least we won't blow up.

Reply:
I like that

Reply:
the other option is to just remove the entire tree, but I'd rather keep it around and create a new account.